sonar-db/src/main/resources/org/sonar/db/component/ComponentMapper.xml

```diff
@@ -250,6 +250,40 @@
     ORDER BY UPPER(p.name), p.name
   </select>
 
+  <select id="selectByQuery" resultType="Component">
+    select
+    <include refid="componentColumns"/>
+    <include refid="sqlSelectByQuery" />
+    ORDER BY LOWER(p.name), p.name
```

Comment:
my point is not about the need for ordering (of course it's required as the results are paged), but about the LOWER conversion.